src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copiedState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(copiedState, action.extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete copiedState[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        for (const key in copiedState) {
+          delete copiedState[key];
+        }
+        break;
+      }
+
+      default: {
+      }
```

Comment:
The `default` case in the switch statement is currently empty. If there are no other action types expected, consider adding a comment to clarify this. Alternatively, if other action types might be added in the future, you could log a warning or throw an error for unrecognized action types.